planner/select.go

```diff
@@ -313,6 +313,11 @@ func findFilteredByIndexedField(scanNode *scanNode) immutable.Option[client.Fiel
 }
 
 func (n *selectNode) initFields(selectReq *mapper.Select) ([]aggregateNode, error) {
+	var isQuerySource bool
+	if n.collection != nil && len(n.collection.Description().Sources) != 0 {
+		_, isQuerySource = n.collection.Description().Sources[0].(*client.QuerySource)
```

Comment:
suggestion: don't you think it makes sense to extract this method to `CollectionDescription`?

It is used in quite some places and [0] indexing is a bit bothering. Would be nice to have it all in one place.

Maybe it should be `GetAllQuerySources` or `HasQuerySource`

Reply:
Would be nice to make the `[0] indexing more documented` or tucked in one place like this suggestion

Reply:
> suggestion: don't you think it makes sense to extract this method to CollectionDescription?

Yeah, I thought about it and then shied away as I didn't want to clutter the public object, is also only actually used in two places I think - there are a couple of other similar but necessarily different places as well that make it look more duplicated than it is.

~~Any function added (e.g. `HasQuerySource`) would really need to be duplicated per type of source.~~
Your `GetAllQuerySources` gave me an idea whilst I wrote that last bit out, I'm probably going to add a `GetSources[Type]() Type` function - thanks!

 - [ ] Add `GetSources[Type]() Type` func
 
 > Would be nice to make the [0] indexing more documented or tucked in one place like this suggestion
 
I'm not sure a comment would add anything here given the name of the variable and the nature of the current code.  The usage of the `isQuerySource` is also documented already.

Reply:
> GetSources[Type]() Type

Golang doesn't actually support this yet...  The func will not be declared on the CollectionDescription object.

Reply:
> Golang doesn't actually support this yet... The func will not be declared on the CollectionDescription object.

I added `func (col CollectionDescription) QuerySources() []*QuerySource` (which calls a private generic func) as workaround, preferring a larger number of funcs on CollectionDescription over a static one floating about in the public package.